src/Razor/src/Microsoft.VisualStudio.LanguageServer.ContainedLanguage/VirtualDocumentBase.cs

```diff
@@ -0,0 +1,91 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.VisualStudio.Text;
+
+namespace Microsoft.VisualStudio.LanguageServer.ContainedLanguage
+{
+    public abstract class VirtualDocumentBase<T> : VirtualDocument where T : VirtualDocumentSnapshot
```

Comment:
Naming seems odd here. VirtualDocumentBase derives from VirtualDocument?

Reply:
I was thinking of maybe DefaultVirtualDocument but I don't have a strong preference. VirtualDocument is semantically is IVirtualDocument interface. I'm going to leave this as is unless someone has a good suggestion and feels strongly about it. Comments, suggestions?

---
In reply to: [450336567](https://github.com/dotnet/aspnetcore-tooling/pull/2174#discussion_r450336567) [](ancestors = 450336567)

Reply:
I like the DefaultVirtualDocument name, can someone from the razor team chime in?

---
In reply to: [450389603](https://github.com/dotnet/aspnetcore-tooling/pull/2174#discussion_r450389603) [](ancestors = 450389603,450336567)

Reply:
I'm not a fan of `DefaultVirtualDocument` only because to me it represents something that can operate on its own (non-abstract). That being said I don't mind `VirtualDocumentBase<T>`

Reply:
Yeah, that seems to be the convention in the codebase I guess, and that's the reason I haven't picked that name to begin with. I'll go ahead and leave it as is.

---
In reply to: [450411863](https://github.com/dotnet/aspnetcore-tooling/pull/2174#discussion_r450411863) [](ancestors = 450411863)

Reply:
Won't fix per comment from Taylor, and I kind of feel the same way. 

---
In reply to: [450409006](https://github.com/dotnet/aspnetcore-tooling/pull/2174#discussion_r450409006) [](ancestors = 450409006,450389603,450336567)